mrp_client.py

```diff
@@ -19,56 +18,64 @@ def __init__(self, parser, args):
 
-    def getmac(self, machine_name, interface_name='eth1'):
-        try:
-            ipgetter = IPGetter(self.urlhandler, machine_name, interface_name)
-            return ipgetter.get_mac()
-        except Exception as err:
-            self.log.fatal(err)
-            exit(1)
 
-    def getnetmask(self, machine_name, interface_name='eth1'):
+    def image_upload(self, command, image_type, desc, arch, path, public, knowngood):
+        image_uploader = ImageControl(self.urlhandler)
         try:
-            ipgetter = IPGetter(self.urlhandler, machine_name, interface_name)
-            return ipgetter.get_netmask()
+            if command == 'upload':
+                rc = image_uploader.upload_image(image_type, desc, arch, path, public,
+                                                 knowngood)
+                self.log.debug(rc)
+            elif command == 'check':
+                if image_uploader.check_for_existence(image_type, desc, arch):
+                    print('True')
+                else:
+                    print('False')
+                    exit(2)
+
         except Exception as err:
             self.log.fatal(err)
             exit(1)
 
+
 if __name__ == '__main__':
     """This is the point of entry of our application, not much logic here"""
-    parser = argparse.ArgumentParser(description='Run some benchmark.')
-    parser.add_argument('action', type=str, default='',
-                        help='getip, getmac, getnetmask')
-    parser.add_argument('machine_name', type=str, default='',
-                        help='name of the machine')
-    parser.add_argument('--mrp-url', type=str, default='',
-                        help='The URL of the MrP server')
-    parser.add_argument('--mrp-token', type=str,
-                        help='The authentication token to use')
+
+    parser = argparse.ArgumentParser(description='Client to the Mr Provisioner \
+                                     server for provisioning baremetal \
+                                     machines.', add_help=False)
     parser.add_argument('-v', '--verbose', action='count', default=0,
                         help='The verbosity of logging output')
-    args = parser.parse_args()
+    parser.add_argument('-u', '--mrp-url', type=str, default='',
+                        required=True, help='The URL of the MrP server')
+    parser.add_argument('-t', '--mrp-token', type=str,
+                        required=True, help='The authentication token to use')
+
+    subparsers = parser.add_subparsers(dest='subcommand')
+
+    parser_image = subparsers.add_parser('image')
+    parser_image.add_argument('-a', '--action', type=str, default='', choices=['check',
+                                                                               'upload'],
+                              required=True, help='Check that an image is on MrP, or upload it.')
+    parser_image.add_argument('-t', '--image-type', type=str, default='',
```

Comment:
Like the benchmark harness, I'd like to keep it simple and not have short versions, or we'll run out of meaningful letters and get confused.

Since this is just a client for use in other scripts, I think that it will be ok.